pkg/idl/listtype_test.go

```diff
@@ -16,40 +16,41 @@ limitations under the License.
 
 package idl_test
 
-// This example shows how you can use the listType map attribute and how
-// to specify a key that will be used to identify specific elements of
-// the list. The listMapKey attribute is used to specify that Name will
-// be used as the key of the map.
+// This example shows how to use the listType map attribute and how to
+// specify a key to identify elements of the list. The listMapKey
+// attribute is used to specify that Name is the key of the map.
 func ExampleListType_map() {
+	type SomeStruct struct {
+		Name  string
+		Value string
+	}
 	type SomeAPI struct {
 		// +listType=map
 		// +listMapKey=name
-		elements []struct {
-			Name string
-
-			Value int
-		}
+		elements []SomeStruct
```

Comment:
I would like more clarity that the tags are there on account of the list type and not on account of the field. Next PR?

Reply:
Let's do next PR so that we don't mix too many changes together.